qemu/tests/qemu_guest_agent.py

```diff
@@ -301,6 +301,83 @@ def gagent_check_halt(self, test, params, env):
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+        """
+        error.context("Check guest agent command 'guest-sync-delimited'",
+                      logging.info)
+        self.gagent.sync("guest-sync-delimited")
+
+    @error.context_aware
+    def _gagent_verify_password(self, vm, new_password, user='root'):
+        """
+        check if the password  works well for the specific user
+        """
+        test_session = vm.wait_for_login(password=new_password)
+
+    @error.context_aware
+    def gagent_check_set_user_password(self, test, params, env):
+        """
+        Execute "guest-set-user-password" command to guest agent
+       :param test: kvm test object
+       :param params: Dictionary with the test parameters
+       :param env: Dictionary with test environment.
+        """
+        old_password = params.get("password", "")
+        new_password = params.get("new_password", "123456")
+        crypted = params.get("crypted", "") == "yes"
+        try:
+            if crypted:
+                self.gagent.set_user_password(new_password, crypted)
+            else:
+                self.gagent.set_user_password(new_password)
+            error.context("check if the guest could be login by new password",
+                          logging.info)
+            test_session = self._gagent_verify_password(self.vm, new_password)
+
+        except guest_agent.VAgentCmdError:
+            raise error.TestError("Failed to set the new password for guest")
+
+        finally:
+            error.context("reset back the password of guest", logging.info)
+            self.gagent.set_user_password(old_password)
+
+    @error.context_aware
+    def gagent_check_get_vcpus(self, test, params, env):
+        """
+        Execute "guest-set-vcpus" command to guest agent
+       :param test: kvm test object
+       :param params: Dictionary with the test parameters
+       :param env: Dictionary with test environment.
+        """
+        self.gagent.get_vcpus()
+
+    @error.context_aware
+    def gagent_check_set_vcpus(self, test, params, env):
+        """
+        Execute "guest-set-vcpus" command to guest agent
+       :param test: kvm test object
+       :param params: Dictionary with the test parameters
+       :param env: Dictionary with test environment.
+        """
+        error.context("get the cpu number of the testing guest")
+        vcpus_info = self.gagent.get_vcpus()
+        vcpus_num = len(vcpus_info)
+        error.context("the vcpu number:%d" % vcpus_num, logging.info)
+        if vcpus_num < 2:
+            raise error.TestNAError("the vpus number of guest should be more"
+                                    " than 1")
+        vcpus_info[vcpus_num - 1]["online"] = False
+        del vcpus_info[vcpus_num - 1]["can-offline"]
+        action = {'vcpus': [vcpus_info[vcpus_num - 1]]}
+        self.gagent.set_vcpus(action)
+
```

Comment:
Hi  @PandaWei , The return check   is done in function cmd() which is called by get_vcpus/set_vcpus,  I think it is enough to handle the return check. what's your opinion?